### ISSUE TITLE
sensu yum repo checksum errors

### ISSUE THREAD

**Issue by elnica6** [sensu/sensu-chef#343]
Hello, I'm receiving the following errors when running the sensu cookbook (v2.8.0):

```
STDERR: http://repos.sensuapp.org/yum/el/6/x86_64/repodata/filelists.xml.gz: [Errno -1] Metadata file does not match checksum
    Trying other mirror.
    http://repos.sensuapp.org/yum/el/6/x86_64/repodata/primary.xml.gz: [Errno -1] Metadata file does not match checksum
    Trying other mirror.

```
here is where the errors are originating from during the chef run:
```
     Resource Declaration:
      ---------------------
      # In /var/chef/cache/cookbooks/yum/providers/repository.rb
      
       55:   execute "yum-makecache-#{new_resource.repositoryid}" do
       56:     command "yum -q makecache --disablerepo=* --enablerepo=#{new_resource.repositoryid}"
       57:     action :nothing
       58:     only_if { new_resource.enabled }
       59:   end
       60: 
      
      Compiled Resource:
      ------------------
      # Declared in /var/chef/cache/cookbooks/yum/providers/repository.rb:55:in `block in class_from_file'
      
      execute("yum-makecache-sensu") do
        action [:nothing]
        retries 0
        retry_delay 2
        guard_interpreter :default
        command "yum -q makecache --disablerepo=* --enablerepo=sensu"
        backup 5
        returns 0
        cookbook_name "sensu"
        only_if { #code block }
      end
      
```
and here as well:

```
    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/sensu/recipes/_linux.rb
    
     62:   repo = yum_repository "sensu" do
     63:     description "sensu monitoring"
     64:     repo = node.sensu.use_unstable_repo ? "yum-unstable" : "yum"
     65:     url "#{node.sensu.yum_repo_url}/#{repo}/el/#{rhel_version_equivalent}/$basearch/"
     66:     action :add
     67:   end
     68:   repo.gpgcheck(false) if repo.respond_to?(:gpgcheck)
    
    Compiled Resource:
    ------------------
    # Declared in /var/chef/cache/cookbooks/sensu/recipes/_linux.rb:62:in `from_file'
    
    yum_repository("sensu") do
      action [:add]
      updated true
      updated_by_last_action true
      retries 0
      retry_delay 2
      guard_interpreter :default
      cookbook_name "sensu"
      recipe_name "_linux"
      description "sensu monitoring"
      baseurl "http://repos.sensuapp.org/yum/el/6/$basearch/"
      repositoryid "sensu"
      mode "0644"
      make_cache true
      enabled true
      sslverify true
    end
```
 basically anytime yum tries to run once the sensu repo has been added. 

I tried removing the sensu repo, `yum clean all` and `yum clean metadata` commands with no luck. Looks like the files have bad checksums...or I'm doing something wrong (which is highly possible lol).

I'm running this on an Amazon Linux EC2 instance running the latest AMI amzn-ami-hvm-2015.03.0.x86_64-gp2.

I also tested on an older version of Amazon Linux amzn-ami-hvm-2014.09.2.x86_64-ebs

Any ideas?

**Comment by elnica6** [comment]
i tested this out by hand as well (following the install instructions from the sensuapp.org site). Pointing to the stable yum repo produces the same error, but pointing to the unstable repo works just fine.

**Comment by elnica6** [comment]
looks like this has been resolved with the latest version update (v2.9.0). improvements to yum repo is listed in change log.

Thanks!
